src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -23,18 +21,17 @@ protected Node(Node<T> prev, T element, Node<T> next) {
 
     @Override
     public void add(T value) {
-        if (size == 0) {
-            first = new Node<>(null, null, null);
-            last = new Node<>(null, null, null);
-            temporaryNode = new Node<>(first, value, last);
-            first = temporaryNode;
-            last = temporaryNode;
-            size++;
-            return;
+        if (size > 0) {
+            Node<T> newNode = new Node<>(last, value, null);
+            last.next = newNode;
+            last = newNode;
+        } else {
+            first = new Node<>(null, value, null);
+            last = new Node<>(null, value, null);
+            Node<T> newNode = new Node<>(last, value, first);
```

Comment:
```suggestion
            Node<T> newNode = new Node<>(null, value, null);
```

Reply:
I have had NPE so i have added one more condition in method unLink()